prediction_market_agent/agents/microchain_agent/functions.py

```diff
@@ -98,44 +95,49 @@ def __call__(self) -> float:
         return balance
 
 
-class BuyYes(Function):
+class BuyTokens(Function):
+    def __init__(self, outcome: str):
+        self.outcome = outcome
+        super().__init__()
+
     @property
     def description(self) -> str:
-        return "Use this function to buy yes outcome tokens of a prediction market. The second parameter specifies how much $ you spend."
+        return f"Use this function to buy {self.outcome} outcome tokens of a prediction market. The second parameter specifies how much $ you spend."
 
     @property
     def example_args(self) -> list[t.Union[str, float]]:
-        return ["Will Joe Biden get reelected in 2024?", 2]
-
-    def __call__(self, market: str, amount: int) -> str:
-        global balance
-        if amount > balance:
-            return (
-                f"Your balance of {balance} $ is not large enough to spend {amount} $."
-            )
-
-        balance -= amount
-        return "Bought " + str(amount * 2) + " yes outcome token of: " + market
-
+        return ["Will Joe Biden get reelected in 2024?", 2.3]
+
+    def __call__(self, market: str, amount: float) -> str:
+        if self.outcome == "yes":
```

Comment:
Outcomes on Omen are `Yes` and `No`, how come it's lowercased here? We have also constants for it:

https://github.com/gnosis/prediction-market-agent-tooling/blob/0e2adab90caf3aa0b73b5e9d126ef81465c5c43e/prediction_market_agent_tooling/markets/omen/data_models.py#L28

Reply:
Good point, fixed.